contracts/src/Teleporter/upgrades/README.md

```diff
@@ -137,18 +127,69 @@ Using specific version:
         ITeleporterMessenger teleporterMessenger = _getTeleporterMessenger();
 ```
 
-#### Receiving from specific Teleporter versions
+### Receiving from specific Teleporter versions
 
 `TeleporterUpgradeable` also provides an initial implementation of [ITeleporterReceiver.receiveTeleporterMessage](../ITeleporterReceiver.sol) that ensures `_msgSender` is a `TeleporterMessenger` contract with a version greater than or equal to `minTeleporterVersion`. This supports the case where a dApp wants to upgrade to a new version of the `TeleporterMessenger` contract, but still wants to be able to receive messages from the old Teleporter contract.The dApp can override `_receiveTeleporterMessage` to implement its own logic for receiving messages from Teleporter contracts.
 
-#### Pausing Teleporter version interactions
+## Managing a TeleporterUpgradeable dApp
+
+dApps that implement `TeleporterUpgradeable` automatically use the latest Teleporter version registered with the `TeleporterRegistry`. Interaction with underlying `TeleporterMessenger` versions can be managed by setting the minimum Teleporter version, and pausing and unpausing specific versions.
+
+### Managing the Minimum Teleporter version
 
-Dapps that inherit from `TeleporterUpgradeable` can pause Teleporter interactions by calling `TeleporterUpgradeable.pauseTeleporterAddress`. This function prevents the contract from interacting with the paused Teleporter address when sending or receiving Teleporter messages.
+The `TeleporterUpgradeable` contract constructor saves the Teleporter registry in a state variable used by the inheriting dApp contract, and initializes a `minTeleporterVersion` to the highest `TeleporterMessenger` version registered in `TeleporterRegistry`. `minTeleporterVersion` is used to allow dApp's to specify the Teleporter versions allowed to interact with it.
+
+#### Updating `minTeleporterVersion`
+
+The `TeleporterUpgradeable.updateMinTeleporterVersion` function updates the `minTeleporterVersion` used to check which Teleporter versions can deliver messages to the dApp, and emits the `MinTeleporterVersionUpdated` event. The `updateMinTeleporterVersion` function should **ONLY** be called by the dApp when it completes delivery of messages from the old Teleporter contract, and now wants to update the `minTeleporterVersion` to only allow the new Teleporter version. By default, `updateMinTeleporterVersion` can only be called with a version greater than the current `minTeleporterVersion` and less than `latestVersion` in the Teleporter registry. So once this function is called, the dApp will no longer be able to receive messages from the old Teleporter contract version, unless the old version's Teleporter address was registered in the registry again with a new version.
+
+> Example: Update the minimum Teleporter version to 2
+>```
+>cast send <DAPP_ADDRESS> "updateMinTeleporterVersion(uint256)" 2 
```

Comment:
should we include other required fields for the command like rpc-url, private key etc as examples

Reply:
Technically those fields aren't strictly required. For example, the rpc url will be taken from the `ETH_RPC_URL` env var if not supplied in the command.

We may still want to include examples with them, though.

Reply:
I added a note to the top of this section that the examples may need to be modified to work with different wallet solutions